### PR TITLE
CreateArtForm

### PR DESCRIPTION
**Pull request by MustafaTRKYLMZ**
createArt form added

### CODE REVIEW

File: libs/ui/src/components/CreateArtForm/CreateArtForm.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ErrorMessage.args = {
+  errorMessage: 'There is a error',
+}
+export const Auth = Template.bind({})
+Auth.args = {
+  auth: {
+    isLoggedIn: true,
+    user: USER_MOCKS[0],
+    token: 'token',
+  },
```
Comment (7alip) [inline libs/ui/src/components/CreateArtForm/CreateArtForm.stories.tsx:47]: We only need `isLoggedIn` in the component.

```suggestion
 isLoggedIn: true
```

File: libs/ui/src/components/CreateArtForm/CreateArtForm.tsx [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      yup.object().shape({
+        label: yup.string(),
+        value: yup.string(),
+      }),
+    ),
+  })
+
+// TODO Consider adding modal form instead of a new page
+export const CreateArtForm: React.FC<CreateArtFormProps> = ({
+  auth,
```
Comment (7alip) [inline libs/ui/src/components/CreateArtForm/CreateArtForm.tsx:57]:
```suggestion
isLoggedIn,
```

File: libs/ui/src/components/CreateArtForm/CreateArtForm.tsx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+                  {t`art.create.require-auth.text`}{' '}
+                  <Navigate href="/user/login" color="blue.500">
+                    {t`art.create.require-auth.button`}
+                  </Navigate>
+                </Text>
+              </VStack>
+            )}
+
+            {/* CREATE FORM */}
+            {auth.isLoggedIn && (
```
Comment (7alip) [inline libs/ui/src/components/CreateArtForm/CreateArtForm.tsx:151]:
```suggestion
            {isLoggedIn && (
```

File: libs/ui/src/components/CreateArtForm/CreateArtForm.tsx [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+                top={0}
+                left={0}
+                boxSize="full"
+                bg="whiteAlpha.900"
+              >
+                <Spinner size="xl" colorScheme="blue" />
+              </Center>
+            )}
+
+            {!auth?.isLoggedIn && (
```
Comment (7alip) [inline libs/ui/src/components/CreateArtForm/CreateArtForm.tsx:139]:
```suggestion
            {!isLoggedIn && (

```

File: libs/ui/src/components/CreateArtForm/CreateArtForm.tsx [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+        </Box>
+        <Box display={{ base: 'none', lg: 'block' }}>{t`art.upload`}</Box>
+      </Button>
+
+      <Modal
+        isCentered
+        closeOnOverlayClick={false}
+        isOpen={formDisclosure.isOpen}
+        onClose={formDisclosure.onClose}
+        size={auth?.isLoggedIn ? '4xl' : 'md'}
```
Comment (7alip) [inline libs/ui/src/components/CreateArtForm/CreateArtForm.tsx:116]:
```suggestion
        size={isLoggedIn ? '4xl' : 'md'}

```

File: libs/ui/src/components/CreateArtForm/types.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  categories: {
+    label: string
+    value: string
+  }[]
+}
+
+export type CreateArtFormProps = {
+  onCreateArt: (data: CreateArtFormFieldValues) => void
+  errorMessage?: string
+  auth?: Auth
```
Comment (7alip) [inline libs/ui/src/components/CreateArtForm/types.tsx:19]:
```suggestion
  isLoggedIn?: boolean

```

File: libs/ui/src/components/CreateArtForm/CreateArtForm.tsx [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+                    name="title"
+                    label={t`title`}
+                    isRequired
+                    errors={errors}
+                    register={register}
+                  />
+                  <WSelect
+                    name="categories"
+                    errors={errors}
+                    control={control as any}
```
Comment (7alip) [inline libs/ui/src/components/CreateArtForm/CreateArtForm.tsx:189]: 1. We can add a comment.
2. We want to be able to select multiple categories, that's why we need to pass `isMulti` prop

```suggestion
                   // TODO: Investigate why ts complains
                    control={control as any}
                    isMulti
```

**Review by 7alip — CHANGES_REQUESTED** [review]
Thank you, we still need to fix some issues.

**Review by 7alip — APPROVED** [review]
Thanks, it's now mergable.

File: libs/ui/src/components/CreateArtForm/CreateArtForm.tsx [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+  useEffect(
+    () => () => {
+      // Make sure to revoke the data uris to avoid memory leaks
+      images.forEach(image => URL.revokeObjectURL((image as any).preview))
+    },
+    [images],
+  )
+
+  const handleCreateArt = async (data: CreateArtFormFieldValues) => {
+    onCreateArt(data)
```
[the remark this answers is not in the document]
Comment (MustafaTRKYLMZ) [reply libs/ui/src/components/CreateArtForm/CreateArtForm.tsx]: I think you added it. Thank you so much

**Pull request merged**